src/components/SQForm/useFormButton.tsx

```diff
@@ -1,25 +1,23 @@
 import React from 'react';
 import {DebouncedState} from 'use-debounce/lib/useDebouncedCallback';
 import {useDebouncedCallback} from 'use-debounce';
-import {useFormikContext} from 'formik';
+import {FormikContextType, useFormikContext} from 'formik';
 import {hasUpdated} from '../../utils';
 
-type UseFormButtonReturnType = {
+type UseFormButtonReturnType<Values> = {
   isButtonDisabled: boolean;
   hasFormBeenUpdated: boolean;
-  values: unknown;
-  initialValues: unknown;
+  initialValues: Values;
   isValid: boolean;
   handleClick: DebouncedState<React.MouseEventHandler<HTMLButtonElement>>;
-  [key: string]: unknown;
-};
+} & FormikContextType<Values>;
 
-export function useFormButton(
+export function useFormButton<Values>(
```

Comment:
With this `Values` generic we can provide `useFormikContext` and `FormikContextType` info about the current values.